### ISSUE TITLE
"C1” in the <<fractional max-pooling>>

### ISSUE THREAD

**Issue by Lighthouseyy** [supercrazysam/Fractional_MaxPooling#2]
Hello, there may be an inappropriate one in your code. Because in the 《fractional max-pooling》paper, the output dimension of fmp is the nearest integer, and int() is rounded down and does not achieve the effect of "ceiling".
Another question is, do you know how the layer of "C1" works in figure 3 of 《fractional max-pooling》?
I hope to hear from you, thank you!

**Comment by OrangeCat95** [comment]
> Hello, there may be an inappropriate one in your code. Because in the 《fractional max-pooling》paper, the output dimension of fmp is the nearest integer, and int() is rounded down and does not achieve the effect of "ceiling".
> Another question is, do you know how the layer of "C1" works in figure 3 of 《fractional max-pooling》?
> I hope to hear from you, thank you!

Can you run this code smoothly？ I have some problem when I was running so can you help me by telling me your version of keras and tensorflow? Thank you a lot.

**Comment by Lighthouseyy** [comment]
Time has passed for a long time, I have forgotten this problem, my mistake is "Input to reshape is a tensor with 98304 values, but the requested shape has 884736", and I have not solved it yet.
I will think again about the solution. If it succeeds, I will reply you!

**Comment by OrangeCat95** [comment]
> Time has passed for a long time, I have forgotten this problem, my mistake is "Input to reshape is a tensor with 98304 values, but the requested shape has 884736", and I have not solved it yet.
> I will think again about the solution. If it succeeds, I will reply you!

Thank you anyway! I will also try to fix it. I will tell you once I make it.

**Comment by OrangeCat95** [comment]
> > Time has passed for a long time, I have forgotten this problem, my mistake is "Input to reshape is a tensor with 98304 values, but the requested shape has 884736", and I have not solved it yet.
> > I will think again about the solution. If it succeeds, I will reply you!
> 
> Thank you anyway! I will also try to fix it. I will tell you once I make it.

Hey, bro. I finally fixed the bug.  Just set seed and seed2 in line 17 of the layer python file equal to 0. All the strange  errors would be disappeared. But I found this CNN model in file cifar_test.py seems not to  converge, which is also totally different of the model proposed in the orginal paper.  Later I will create a new rep in my github. If you are interested in this paper and it's implementation in Keras, welcome to pay attention!

**Comment by Lighthouseyy** [comment]
I got the same result as you, and the model didn't converge.
I have tried to experiment on the cifar100 dataset using the methods in the original paper. However, I don't understand the meaning of C2 and C1. I analyze C2 as "2×2 size, the stride is 1, with valid padding" convolution operation . Then C1 should be a convolution operation with a size of 1×1 and a stride of 1?

**Comment by OrangeCat95** [comment]
> I got the same result as you, and the model didn't converge.
> I have tried to experiment on the cifar100 dataset using the methods in the original paper. However, I don't understand the meaning of C2 and C1. I analyze C2 as "2×2 size, the stride is 1, with valid padding" convolution operation . Then C1 should be a convolution operation with a size of 1×1 and a stride of 1?

I think you can read another paper "Spatially-sparse convolutional neural networks" authored by Graham, etc. The notatation used in FMP paper is same in that paper. Your idea on C1 and C2 is partly right. What's more, you should also pay attention to the # filters in C2 and C1.

**Comment by OrangeCat95** [comment]
> > I got the same result as you, and the model didn't converge.
> > I have tried to experiment on the cifar100 dataset using the methods in the original paper. However, I don't understand the meaning of C2 and C1. I analyze C2 as "2×2 size, the stride is 1, with valid padding" convolution operation . Then C1 should be a convolution operation with a size of 1×1 and a stride of 1?
> 
> I think you can read another paper "Spatially-sparse convolutional neural networks" authored by Graham, etc. The notatation used in FMP paper is same in that paper. Your idea on C1 and C2 is partly right. What's more, you should also pay attention to the # filters in C2 and C1.

By the way, can you give me your e-mail address? There are some other problems in this paper and its implementation I want to discuss with you. Thank you very much!

**Comment by Lighthouseyy** [comment]
Sure, thank you for your reply about “C1”！
You can send e-mail to chwang_cauc@163.com. Hope we can gain something!
